### PR TITLE
DOC: update contributing to pandas documentation

### LINKED ISSUE #48035: DOC: "Committing your code" in "Contributing to pandas" needs update
### Pandas version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://pandas.pydata.org/docs/development/contributing.html#committing-your-code

### Documentation problem

In the "Committing your code" section of the "Contributing to pandas" documentation there are two problems. 

However, I don't see that this convention is still in use. It seems to be common to reference the respective issue in the PR description, such as "closes #1234".

### 2. It is further said: 

> Either style is fine, but the former is generally preferred:
> 
> - a subject line with < 80 chars.
> - One blank line.
> - Optionally, a commit message body.
> 
> Now you can commit your changes in your local repository:
> 
> `git commit -m`

The command `git commit -m` raises an error that no commit message is given. There are two possibilities here:

1. Change the command to `git commit -m "commit message comes here"`, which users can do to have a single line commit message. It is possible to have a multi line commit message this way, but this doesn't seem to be used frequently.
2. Change the command to `git commit` which opens an editor for the user where they can write a multiline commit message as described above. However, since the project seems to almost always use "squash and merge", I'm not sure how those commit messages are handled. 

### Suggested fix for documentation

For issue No 1: 
Since the convention of including the respective GitHub issue in the commit message doesn't seem to be in use anymore, the documentation for this could be removed. 

For issue No 2:
It would be nice to make the documentation of how the commit messages should be written/ structured consistent with the conventions of the core developers.

I'm happy to update the documentation myself after decisions on how to proceed have been made.

### PR DESCRIPTION
**Pull request by marenwestermann**
- [x] closes #48035
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

I removed the part in the [Committing your code](https://pandas.pydata.org/docs/development/contributing.html#id9) section where it is said that commit messages should include the number of the GitHub issue the commit is contributing towards. I updated the pull request template to include cross references so that it's more clear how people should make use of referencing GitHub issues.

I fixed the code snippet `git commit -m` which causes an error and updated the documentation to include both commit message versions, the one line commit message and the more detailed commit message that can be done with an editor. Let me know if you would like to keep both versions or would like further changes.

Additionally, I moved the section about the prefixes to the [Finally, make the pull request](https://pandas.pydata.org/docs/development/contributing.html#id12) section of the documentation. Let me know if the prefixes need to be updated.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
 - [ ] closes #xxxx (Replace xxxx with the Github issue number)
+- [ ] xref #xxxx (Replace xxxx with the Github issue number)
```
Comment (mroeschke) [inline .github/PULL_REQUEST_TEMPLATE.md:2]: Why was this added? I am concern that it may be confusing with there also being `closes #xxxx above`

IMO we want to encourage the users to use `closes #xxxx` if applicable so it automatically closes the linked issue

Comment (phofl) [reply .github/PULL_REQUEST_TEMPLATE.md:2]: Yeah agreed. There are already many checkboxes you have to tick, would prefer to not add this

File: doc/source/development/contributing.rst [hunk 2]
```diff
@@ -194,31 +194,18 @@ Doing 'git status' again should give something like::
...
-* BLD: Updates to the build process/scripts
-* PERF: Performance improvement
-* TYP: Type annotations
-* CLN: Code cleanup
-
-The following defines how a commit message should be structured.  Please reference the
-relevant GitHub issues in your commit message using GH1234 or #1234.  Either style
-is fine, but the former is generally preferred:
+Finally, commit your changes to your local repository with an explanatory commit
+message of ``< 80`` chars::
```
Comment (datapythonista) [inline doc/source/development/contributing.rst:198]:
```suggestion
message::
```
I know this already exists now, but since we squash commits, not very important, and we can simplify things a bit.

File: doc/source/development/contributing.rst [hunk 3]
```diff
@@ -194,31 +194,18 @@ Doing 'git status' again should give something like::
...
+
+    git commit -m "your commit message goes here"
+
+Alternatively, you can just type ``git commit`` which opens an editor and use the
+following commit message structure:
 
 * a subject line with ``< 80`` chars.
 * One blank line.
 * Optionally, a commit message body.
```
Comment (datapythonista) [inline doc/source/development/contributing.rst:207]: I'd just delete all this.

**Review by datapythonista — COMMENTED** [review]
Thanks @marenwestermann, nice update. Added couple of comments of things that I'd change. In the current docs, your changes look fine other than the template changes already mentioned.

**Comment by marenwestermann** [comment]
Thanks for your feedback! I addressed your comments. I added the line `- [ ] xref #xxxx (Replace xxxx with the Github issue number)` in `PULL_REQUEST_TEMPLATE.md` because sometimes a pull request contributes to an issue but doesn't close it such as here: https://github.com/pandas-dev/pandas/pull/44798. However, I understand your concerns and removed this line.

**Review by datapythonista — APPROVED** [review]
lgtm, thanks @marenwestermann, nice improvement.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @marenwestermann!
